src/Metabor/Semaphore/LockAdapterMutex.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Metabor\Semaphore;
+
+use MetaborStd\Semaphore\LockAdapterInterface;
+use MetaborStd\Semaphore\MutexInterface;
+
+class LockAdapterMutex implements MutexInterface
+{
+    /**
+     * @var LockAdapterInterface
+     */
+    private $lockAdapter;
+
+    /**
+     * @var string
+     */
+    private $resourceName;
+
+    /**
+     * @var bool
+     */
+    private $acquired = false;
+
+    /**
+     * @param LockAdapterInterface $lockAdapter
+     * @param string               $resourceName
+     */
+    public function __construct(LockAdapterInterface $lockAdapter, $resourceName)
+    {
+        $this->lockAdapter = $lockAdapter;
+        $this->resourceName = $resourceName;
+    }
+
+    /**
+     * @see \MetaborStd\Semaphore\MutexInterface::releaseLock()
+     */
+    public function releaseLock()
+    {
+        if ($this->acquired) {
+            $result = $this->lockAdapter->releaseLock($this->resourceName);
+            if ($result) {
+                $this->acquired = false;
+            }
+
+            return $result;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @see \MetaborStd\Semaphore\MutexInterface::isAcquired()
+     */
+    public function isAcquired()
+    {
+        return $this->acquired;
+    }
+
+    /**
+     * @see \MetaborStd\Semaphore\MutexInterface::acquireLock()
+     */
+    public function acquireLock()
+    {
+        if (!$this->acquired) {
+            $this->acquired = $this->lockAdapter->acquireLock($this->resourceName);
```

Comment:
Maybe its a good idea that LockAdapterInterface requires timeout value in acquireLock method as the sm process might die without releasing the lock. We can of course require this value in the constructor of a class implementing LockAdapterInterface or having it as dependency of LockAdapterMutex